src/fsharp/FSharp.Compiler.LanguageServer/LspTypes.fs

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace FSharp.Compiler.LanguageServer
+
+// Types as defined at https://microsoft.github.io/language-server-protocol/specification
+
+/// Represents a zero-based line and column of a text document.
+type Position = {
+    line: int
+    character: int
+}
+
+type Range = {
+    start: Position
+    ``end``: Position
+}
+
+type DocumentUri = string
+
+type Location = {
+    uri: DocumentUri
+    range: Range
+}
+
+type DiagnosticRelatedInformation = {
+    location: Location
+    message: string
+}
+
+type Diagnostic = {
```

Comment:
Note: you may want to go with

```
type Foo
    {
        Label: hoopty
    }
    with
        static member Hober = "banana"
```

Because this is pretty much the only format that lets you indent/de-indent without the compiler freaking out.

Reply:
Done.